src/export/general/generalExport.go

```diff
@@ -177,15 +177,18 @@ func ExportJSON(filename string, iter uint32, refreshRate uint64) error {
 
 	encoder := json.NewEncoder(logFile)
 	stats := NewOverallStats()
-	var i uint32
 
-	for i = 0; i < iter; i++ {
+	for i := uint32(0); i < iter; i++ {
 		err := stats.updateData()
 		if err != nil {
-			return err
+			fmt.Println(err)
 		}
 
 		err = encoder.Encode(&stats)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
`continue` here as well?

Reply:
The only statements after are the sleep, it would be better to sleep after an error occurred than immediately attempt to refecth I felt. Even in the previous case, it would be better to sleep, but we can skip the second block if the first fails. Would that suffice?